terraform/main.tf

```diff
@@ -0,0 +1,61 @@
+# Enable Google Cloud APIs
+module "enable_google_apis" {
+  source                      = "terraform-google-modules/project-factory/google//modules/project_services"
+  version                     = "~> 13.0"
+
+  project_id                  = var.gcp_project_id
+  disable_services_on_destroy = false
+
+  activate_apis               = var.apis
+}
+
+# Create GKE Cluster
+resource "google_container_cluster" "cluster" {
+  name               = var.name
+  location           = var.region
+  ip_allocation_policy {
+  }
+
+  # Enabling Autopilot for this cluster
+  enable_autopilot = true
+
+  depends_on = [
+    module.enable_google_apis
+  ]
+}
+
+# Get Credentials for Cluster
+resource "null_resource" "get_credentials" {
```

Comment:
use of `gke_auth` module instead of a local exec maybe?

Reply:
nit: it is generally advised to avoid using `null_resource` because it does not have a clear destroy path. in this particular it can be OK to use it. however, you should take care that credentials used by `gcloud` will have the same authZ level as the credentials used by the terraform to provision resources using `gcp` provider.

in addition to `gke_auth` module, you can look at [terraform-google-gcloud](https://github.com/terraform-google-modules/terraform-google-gcloud). There are other options as well.